src/lib/router.ts

```diff
@@ -19,12 +20,60 @@ router.post('/matches', async (req, res, next) => {
       highlights: [],
     };
 
-    const inserted = await getMatchesCollection().insertOne(match);
+    const matchesCollection = await getMatchesCollection();
+    const inserted = await matchesCollection.insertOne(match);
     if (!inserted.acknowledged) {
-      res.status(500).send('Error inserting match');
+      response.status(500).send('Error inserting match');
       return;
     }
-    res.status(200).json(match);
+    response.status(200).json(match);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.post('/matches/:id/highlights', async (request, response, next) => {
+  try {
+    const { timestamp, type, videoSrc } = request.body;
+    const { id } = request.params;
+    const requestIsInvalid =
+      typeof timestamp !== 'number' ||
+      typeof type !== 'string' ||
+      typeof videoSrc !== 'string';
+
+    if (requestIsInvalid) {
+      response.status(400).send('Invalid payload');
+      return;
+    }
+
+    const highlight: MatchHighlight = {
+      timestamp,
+      type,
+      videoSrc,
+    };
+
+    const matchesCollection = await getMatchesCollection();
+    const inserted = await matchesCollection.updateOne(
+      { _id: ObjectId.createFromHexString(id) },
```

Comment:
There is no validation if `id` is a valid `ObjectId` string.
Thus, you could call `/api/matches/NOT_A_OBJECT_ID/highlights` and run into an exception thrown by `ObjectId.createFromHexString` (or it returns something invalid).